rethinkdb/datadog_checks/rethinkdb/connections.py

```diff
@@ -0,0 +1,83 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+"""
+RethinkDB connection interface and implementations.
+"""
+from typing import Any, TypedDict
+
+import rethinkdb
+
+# See: https://rethinkdb.com/api/python/server
+ConnectionServer = TypedDict('ConnectionServer', {'id': str, 'name': str, 'proxy': bool})
+
+ConnectionTags = TypedDict('ConnectionTags', {'server': str, 'host': str, 'port': int, 'proxy': bool})
+
+
+class Connection(object):
```

Comment:
Same here, don't really see the point.

Reply:
Similar answer to: https://github.com/DataDog/integrations-core/pull/5715#discussion_r394209278, a) increased testability (eg I replace this with a fake implementation [here](https://github.com/DataDog/integrations-core/pull/5715/files#diff-a6bd6e5ef8594de4414bf5cf7f3a72d3R52-R56)), b) less coupling with the details of the `rethinkdb` library. (Also c) improved type information)

Reply:
About testability, you can easily achieve the same result using mock/path from `mock` library. Mocking that way, is usually shorter than subclassing.

Reply:
Addressed in c64e1e5e7. I still think the resulting code is less _correct_, but I want us to make progress here. (Having to mock our own code generally indicates code smells IMO. I didn't _have_ to here, but now the fake connection used for unit tests is bound the the `rethinkdb` client library.)

Reply:
I would generally agree but given the small scope of individual integrations and that we know they're not systems that are going to evolve a lot over time (vs any other type of software) I think they call for simplicity (straightfowardness) over extensibility/encapsulation etc